src/index.ts

```diff
@@ -214,7 +215,7 @@ class Molecule {
       case 'option':
         return deserializeOption(value)
       case 'union':
-        return deserializeUnion(value, [littleHexToInt(value.slice(0, HEADER_ELEMENT_SIZE))])
+        return deserializeUnion(value, [littleHexToInt(value.slice(0, HEADER_ELEMENT_SIZE * 2))])
```

Comment:
@duanyytop the second param derives from the first, so is it possible to remove the second param?

Reply:
Yes. We can remove the second param

Reply:
I update it in this pr or you update ?

Reply:
> I update it in this pr or you update ?

Please submit a PR to this branch.

Reply:
OK

Reply:
> OK

And I found another problem and working on it now, I'll set this PR to HOLD until I fix it.

Reply:
> > OK
> 
> And I found another problem and working on it now, I'll set this PR to HOLD until I fix it.

😂 It's a wrong test case so the code is OK

Reply:
This update from `HEADER_ELEMENT_SIZE` to `HEADER_ELEMENT_SIZE * 2` is wrong according to the README, it's a 32-bit number, please fix the update in your PR

Reply:
OK. I think we should check the index of union serialized data and items length. I will update this logic.

Reply:
> OK. I think we should check the index of union serialized data and items length. I will update this logic.

Merged